R/utils.R

```diff
@@ -36,3 +36,85 @@ calc_disc_dist_quantile <- function(prob, days, quantile) {
   names(quantiles) <- as.character(quantile)
   quantiles
 }
+#' Creates a JSON version of the epiparameter database
+#'
+#' @description This converts the database from a data frame (read from the csv
+#' stored in the package) into a JSON data file and saves it. This functions
+#' allows the database to be edited and then the JSON file can be updated but
+#' calling this function.
+#'
+#' @param read_path A relative file path from the package root directory to read
+#' the parameters database from
+#' @param write_path A relative file path from the package root directory to
+#' write the JSON data file to
+#'
+#' @return Invisibly returns the JSON data
+#' @keywords internal
+make_json_data <- function(read_path = file.path("extdata", "parameters.csv"),
+                           write_path = file.path("inst", "extdata", "data.json")) {
+  # read in epiparameter database
+  data <- utils::read.csv(
+    file = system.file(
+      read_path,
+      package = "epiparameter",
+      mustWork = TRUE
+    ), header = TRUE
+  )
+
+  # convert cells with arrays to numeric
+  for (i in grep(pattern = "_ci$", x = colnames(data))) {
+    if (!all(is.na(data[[i]]))) {
+      data[[i]] <- lapply(
+        strsplit(x = data[[i]], split = ",", fixed = TRUE),
+        as.numeric
+      )
+    }
+
+  }
+
+  # convert data to json
+  data_json <- jsonlite::toJSON(data, dataframe = "columns", na = "null")
+  jsonlite::validate(data_json)
+
+  # save json data
+  jsonlite::write_json(data_json, path = write_path)
+
+  # invisibly return json data
+  invisible(data_json)
+}
```

Comment:
I think `make_json_data()` is a useful overall, but I would restructure it to avoid using `jsonlite::validate()` and `jsonlite::toJSON()` - I don't think either of them are necessary. It's possible to pass the dataframe to `jsonlite::write_json()` directly, and I would suggest passing the option `pretty = TRUE` to write out a human-readable JSON file. As I understand it, the use of `toJSON` with `write_json` is not correct, see https://stackoverflow.com/questions/69617597/importing-editing-and-saving-json-in-a-simple-way.

Reply:
Seems resolved in https://github.com/epiverse-trace/epiparameter/commit/ffff318e5c0f595a2e4a3b6666832457f931a866